src/rust/sysmon_parser/src/error.rs

```diff
@@ -0,0 +1,30 @@
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Represents all possible errors that can occur when parsing Sysmon XML events.
+#[derive(Error, Debug, PartialEq)]
```

Comment:
https://blog.rust-lang.org/2019/12/19/Rust-1.40.0.html#non_exhaustive-structs-enums-and-variants

Reply:
TIL this looks like what I want in other places too.